dashboard/app/views/levels/editors/_gamelab.html.haml

```diff
@@ -93,6 +93,12 @@
   = f.label 'Custom setup code'
   = f.text_field :custom_setup_code
 
+.field
+  = f.label 'Validation code'
+  %p
+    This is a snippet of javascript that is run after every draw loop. Call levelSuccess() in here to stop the level as a success. You can also pass a number into levelSuccess() to give the student a non-perfect test result (see <a href="https://github.com/code-dot-org/code-dot-org/blob/101fbfd2dcdd635d7359e991559ba782f9fd00be/apps/src/constants.js#L33-L106">TestResults</a> for the list of valid test results).
+  = f.text_area :validation_code
```

Comment:
What is our API contract with level builders?  If they depend on internal behavior, and we change internal behavior, who owns fixing?

Who is responsible for keeping the code working, and testing it after deploys?

Reply:
I'm not sure, but I have a couple thoughts:
* Ideally levelbuilders will only use standard Gamelab properties and methods (which are stable anyway), and things specifically provided by `ValidationSetup`. We can add accessors to `ValidationSetup` for things exposed by the helper library for when levelbuilders need to inspect things like behaviors.
* Unless we programmatically prevent access to helper library methods and properties, I suspect levelbuilders will wind up using them anyway, and my instinct is that it would be very difficult to stop them programmatically.